File: erp/src/main/java/com/erp/model/SalesPayment.java
package com.erp.model;

import javax.persistence.*;
import java.util.Date;

@Entity(name = "sales_payment")
@Table(name = "sales_payment")
public class SalesPayment {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column(name = "sales_payment_code")
    private Long salesPaymentCode;

    @Column(name = "customer_code")
    private Long customerCode;

    @Column(name = "customer_name")
    private String customerName;

    @Column(name = "sales_order_code")
    private Long salesOrderCode;

    @Column(name = "amount_paid")
    private Double amountPaid;

    @Column(name = "pay_date")
    @Temporal(TemporalType.DATE)
    private Date payDate;

    @Column(name = "payment_mode")
    private String paymentMode;

    @Column(name = "payment_ref")
    private String paymentRef;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Long getSalesPaymentCode() {
        return salesPaymentCode;
    }

    public void setSalesPaymentCode(Long salesPaymentCode) {
        this.salesPaymentCode = salesPaymentCode;
    }

    public Long getCustomerCode() {
        return customerCode;
    }

    public void setCustomerCode(Long customerCode) {
        this.customerCode = customerCode;
    }

    public String getCustomerName() {
        return customerName;
    }

    public void setCustomerName(String customerName) {
        this.customerName = customerName;
    }

    public Long getSalesOrderCode() {
        return salesOrderCode;
    }

    public void setSalesOrderCode(Long salesOrderCode) {
        this.salesOrderCode = salesOrderCode;
    }

    public Double getAmountPaid() {
        return amountPaid;
    }

    public void setAmountPaid(Double amountPaid) {
        this.amountPaid = amountPaid;
    }

    public Date getPayDate() {
        return payDate;
    }

    public void setPayDate(Date payDate) {
        this.payDate = payDate;
    }

    public String getPaymentMode() {
        return paymentMode;
    }

    public void setPaymentMode(String paymentMode) {
        this.paymentMode = paymentMode;
    }

    public String getPaymentRef() {
        return paymentRef;
    }

    public void setPaymentRef(String paymentRef) {
        this.paymentRef = paymentRef;
    }
    
}

File: ebs_flutter/lib/model/SalesPayment.dart
class SalesPayment {
  final int? id;
  final int? salesPaymentCode;
  final int? customerCode;
  final String? customerName;
  final int? salesOrderCode;
  final double? amountPaid;
  final DateTime? payDate;
  final String? paymentMode;
  final String? paymentRef;

  SalesPayment({
    this.id,
    this.salesPaymentCode,
    this.customerCode,
    this.customerName,
    this.salesOrderCode,
    this.amountPaid,
    this.payDate,
    this.paymentMode,
    this.paymentRef,
  });

  factory SalesPayment.fromJson(Map<String, dynamic> json) {
    return SalesPayment(
      id: json['id'],
      salesPaymentCode: json['salesPaymentCode'],
      customerCode: json['customerCode'],
      customerName: json['customerName'],
      salesOrderCode: json['salesOrderCode'],
      amountPaid: json['amountPaid'] != null ? (json['amountPaid'] as num).toDouble() : null,
      payDate: json['payDate'] != null ? DateTime.parse(json['payDate']) : null,
      paymentMode: json['paymentMode'],
      paymentRef: json['paymentRef'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'salesPaymentCode': salesPaymentCode,
      'customerCode': customerCode,
      'customerName': customerName,
      'salesOrderCode': salesOrderCode,
      'amountPaid': amountPaid,
      'payDate': payDate?.toIso8601String(),
      'paymentMode': paymentMode,
      'paymentRef': paymentRef,
    };
  }
}